site-src/concepts/guidelines.md

```diff
@@ -118,6 +140,18 @@ The "implementation-specific" designation allows a CORE or EXTENDED feature to
 be well-defined taking into account the realities of some features that are
 mostly but not entirely portable.
 
+## Limitations of Webhook and CRD Validation
+CRD and webhook validation is not the final validation i.e. webhook is "nice UX"
+but not schema enforcement. This validation is intended to provide immediate
+feedback to users when they provide an invalid configuration. Write code
+defensively with the assumption that at least some invalid input (Gateway API
+resources) will reach your controller. Both Webhook and CRD validation is not
+fully reliable because it:
+
+* Is not guaranteed to be present or up to date in all clusters.
```

Comment:
I think that if we mandate that a Gateway API install _must_ include either the webhook or something that implements the same function, this can be removed.

Reply:
Good call, replaced with "May not be deployed correctly"